retrieval/scrape.go

```diff
@@ -65,11 +65,30 @@ var (
 		},
 		[]string{interval},
 	)
+	targetScrapeFailures = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "target_scrape_failures_total",
+			Help:      "Total number of scrapes that failed due to HTTP failure.",
+		},
+		[]string{interval},
```

Comment:
I'd skip interval, status maybe, but probably not too helpful.

The scraped job might be interesting for several of these metrics.